base/reduce.jl

```diff
@@ -579,11 +569,12 @@ true
 ```
 """
 any(f::Any, itr) = any(Predicate(f), itr)
-any(f::Predicate, itr) = mapreduce_sc_impl(f, |, itr)
-any(f::typeof(identity), itr) =
-    eltype(itr) <: Bool ?
-        mapreduce_sc_impl(f, |, itr) :
-        reduce(or_bool_only, false, itr)
+function any(f::Predicate, itr)
```

Comment:
You dont' need this, just leave

```julia
any(f::Predicate, itr) = mapreduce_sc_impl(f, |, itr)
all(f::Predicate, itr) = mapreduce_sc_impl(f, &, itr)
```

(as the `mapreduce_sc_impl` definition is effectively the same as the loops) and remove the specializations for `identity`.